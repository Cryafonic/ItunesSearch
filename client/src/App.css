/* main page. */
.favButton {
  margin: 0.5% 0 0.1% 2%;
}

.card {
  margin: 0.3% 0% 0.3% 2%;
  width: 30%;
}
/* when viewing the favourites page. */

h1 {
  position: absolute;
  margin: 0 0 0 40%;
  border-bottom: lightgray solid 1px;
}

.favPageButton {
  margin: 2%;
  margin-top: 4.7%;
}

.underline {
  border-bottom: lightgray solid 1px;
}

/* @media queries */

@media only screen and (max-width: 1500px) {
  .card {
    margin: 0.3% 0% 0.3% 2%;
    width: 35%;
  }
}

@media only screen and (max-width: 1030px) {
  .card {
    margin: 0.3% auto 0.3% auto;
    width: 80%;
  }
  .favButton {
    margin: 0 0 0.1% 10%;
  }
}