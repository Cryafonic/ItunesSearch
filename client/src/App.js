import FromInput from "./components/formInput/FromInput";
import 'bootstrap/dist/css/bootstrap.min.css';
import { useEffect, useState } from "react";
import { FetchOnURL } from "./components/fetchMethods/FetchOnURL";
import Loading from "./components/loading/Loading";
import Modal from 'react-bootstrap/Modal';
import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';


// TODO: fix refresh issue

function App() {
  const [itunesData, setItunesData] = useState(null);
  const [loaded, setLoaded] = useState(false);
  const [show, setShow] = useState(false);

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  useEffect(() => {
    if  (!loaded){
      FetchOnURL("/search/results").then(res => {
          if (Object.keys(res).length !== 0) {
            setItunesData(res);
            setLoaded(true);
          } else {
            setLoaded(false);
          }
      });
    }
  },[itunesData]);
  
  if (!loaded){
    return(
      <Loading />
    );
  }else {
    return(
      <>
        <Modal show={show} onHide={handleClose}>
          <Button variant="primary" onClick={handleClose}>
            close
          </Button>
          {/* TODO: map over all the objects that have been added to the array and been removed. */}
        </Modal>
        {/* TODO: style input field */}
        <FromInput />
        <Button variant="primary" onClick={handleShow}>
          View Favourites
        </Button>
          {
            itunesData.map((item, key) => {
             return(
              <Card key={key}>
                <Card.Body>
                        <img
                          src={item.artworkUrl100}
                          height="100px"
                          width="100px"
                        />
                        <Card.Title>{item.artistName}</Card.Title>
                        <Card.Text className="align-left">{item.collectionName}</Card.Text>
                        {/* TODO: create add to fav function */}
                        <Button variant="primary" >Add To Favourites</Button>
                </Card.Body>
              </Card>
             );
            })
          }
      </>
    );
  };
}

export default App;
