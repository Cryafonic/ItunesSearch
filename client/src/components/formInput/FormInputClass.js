import React, { Component } from "react";
import { FetchParams } from "../fetchMethods/FetchParams";
import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';
import "../formInput/formInputClass.css";

export default class FormInputClass extends Component{
    state = {
        input: '',
        select: 'all'
    }

    // handles the value when it changes
    handleValueChange = (e) => {
        const value = e.target.value
        this.setState({
            ...this.state,
            [e.target.name]: value
        })
    }

    // lifting state up to be access on the top level.
    handleSubmit = (e) => {
        e.preventDefault();
        this.props.ituneSearch(this.state);
        this.setState({
            input: ''
        });
    }

    // renders the input and dropdown to the DOM.
    render() {
        return (
            <form onSubmit={this.handleSubmit} >

                        <input
                            className="search"
                            type="text" 
                            name="input"
                            value={this.state.input}
                            onChange={this.handleValueChange}
                            placeholder="Search"
                            required
                        />

                        <select className="dropDown" onChange={this.handleValueChange} name="select">
                            <option value="all" >All</option>
                            <option value="music" >Music</option>
                            <option value="movie" >Movie</option>
                            <option value="podcast" >Podcast</option>
                            <option value="musicVideo" >Music Video</option>
                            <option value="audiobook" >Audio book</option>
                            <option value="shortFilm" >Short Film</option>
                            <option value="tvShow" >Tv Show</option>
                            <option value="software" >Software</option>
                            <option value="ebook" >Ebook</option>
                        </select>
                        <Button
                            type="submit"
                            variant="success"
                            value="Search"
                        >Search</Button>
            </form>
        );
    }
}