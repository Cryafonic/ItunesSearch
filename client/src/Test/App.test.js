import {cleanup, React, render, renderer} from "@testing-library/react";
import App from '../App';
import FormInputClass from "../components/formInput/FormInputClass";
import {mount, shallow} from 'enzyme';
// import { expect } from "chai";

afterEach(cleanup);

describe("Tests for app.js" , () => {
  it("checking for unexpected changes.", () => {
    const wrapper = render(<App />)
    expect(wrapper).toMatchSnapshot();
  })

  it("testing if components match is falsy", async () => {
    const wrapper = render(<FormInputClass />);
    const wrapper1 = render(<App />);
    expect(wrapper).not.toBe(wrapper1);
  });

  it("test's if the input handles change events when given a new value", () => {
    const wrapper = shallow(<FormInputClass />);
    wrapper.find('input').simulate("change",{
      target: { input: "Hello"}
    });
  });
})


