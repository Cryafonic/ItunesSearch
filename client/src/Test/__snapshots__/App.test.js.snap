// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for app.js checking for unexpected changes. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <input
          class="search"
          name="input"
          placeholder="Search"
          required=""
          type="text"
          value=""
        />
        <select
          class="dropDown"
          name="select"
        >
          <option
            value="all"
          >
            All
          </option>
          <option
            value="music"
          >
            Music
          </option>
          <option
            value="movie"
          >
            Movie
          </option>
          <option
            value="podcast"
          >
            Podcast
          </option>
          <option
            value="musicVideo"
          >
            Music Video
          </option>
          <option
            value="audiobook"
          >
            Audio book
          </option>
          <option
            value="shortFilm"
          >
            Short Film
          </option>
          <option
            value="tvShow"
          >
            Tv Show
          </option>
          <option
            value="software"
          >
            Software
          </option>
          <option
            value="ebook"
          >
            Ebook
          </option>
        </select>
        <button
          class="btn btn-success"
          type="submit"
          value="Search"
        >
          Search
        </button>
      </form>
      <div>
        <h2
          class="loading"
        >
          Loading...
        </h2>
        <div
          class="spinner spinner-border text-success"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <form>
      <input
        class="search"
        name="input"
        placeholder="Search"
        required=""
        type="text"
        value=""
      />
      <select
        class="dropDown"
        name="select"
      >
        <option
          value="all"
        >
          All
        </option>
        <option
          value="music"
        >
          Music
        </option>
        <option
          value="movie"
        >
          Movie
        </option>
        <option
          value="podcast"
        >
          Podcast
        </option>
        <option
          value="musicVideo"
        >
          Music Video
        </option>
        <option
          value="audiobook"
        >
          Audio book
        </option>
        <option
          value="shortFilm"
        >
          Short Film
        </option>
        <option
          value="tvShow"
        >
          Tv Show
        </option>
        <option
          value="software"
        >
          Software
        </option>
        <option
          value="ebook"
        >
          Ebook
        </option>
      </select>
      <button
        class="btn btn-success"
        type="submit"
        value="Search"
      >
        Search
      </button>
    </form>
    <div>
      <h2
        class="loading"
      >
        Loading...
      </h2>
      <div
        class="spinner spinner-border text-success"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
